specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -16717,6 +16717,21 @@
           "description": "IP allocation type for recovery VM.",
           "type": "string"
         },
+        "recoveryPublicIpAddressId": {
```

Comment:
>recoveryPublicIpAddressId [](start = 9, length = 25)

You are modifying a stable API version by adding extra properties, which is considered a breaking change. These types of changes should be introduced in a new API version.

Reply:
As per [api-guidelines/definition-of-breaking-change](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change), these changes do not qualify as a breaking change. Changes aren't made to the existing parameters and neither are these mandatory. Hence, there's no issue related to the backward compatibility.

Reply:
IMHO, it only applies for `required` properties.

Reply:
Yeah, I missed the `required` aspect and assumed the worst. Sorry for the confusion.

---
In reply to: [296965293](https://github.com/Azure/azure-rest-api-specs/pull/6380#discussion_r296965293) [](ancestors = 296965293)